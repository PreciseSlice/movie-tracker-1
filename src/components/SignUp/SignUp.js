import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addUser } from '../../actions/index';
import { fetchUser } from '../../helpers/helper';

class SignUp extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      password: '',
      email: ''
    };
  }

  async addUserDatabase(event) {
    event.preventDefault();
    const { name, password, email } = this.state;

    try {
      const getUserData = await fetchUser();

      const match = getUserData.data.find(user => {
        // Should we return true or false instead of the object or undefined?
        // It works for now, is there a better way?
        return user.email === email;
      });
      if (!match) {
        await fetch('/api/users/new', {
          method: 'POST',
          body: JSON.stringify({ name, password, email }),
          headers: {
            'Content-Type': 'application/json'
          }
        });
      } else {
        return alert('Email allready registered!');
      }
    } catch (error) {
      const error = new Error('addUser failed to add user');
      return error;
    }
  }

  render() {
    const { handleSubmit } = this.props;

    return (
      <section>
        <form
          onSubmit={event => {
            event.preventDefault();
            handleSubmit(
              this.state.name,
              this.state.email.toLowerCase(),
              this.state.password
            );
            this.addUserDatabase(event);
          }}
        >
          <input
            value={this.state.name}
            placeholder="name"
            autoComplete="email"
            onChange={event => this.setState({ name: event.target.value })}
            required
          />
          <input
            value={this.state.email}
            placeholder="email"
            type="email"
            autoComplete="email"
            onChange={event => this.setState({ email: event.target.value })}
            required
          />
          <input
            value={this.state.password}
            placeholder="password"
            type="password"
            autoComplete="on"
            onChange={event => this.setState({ password: event.target.value })}
            required
          />
          <button>Sign Up</button>
        </form>
      </section>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  handleSubmit: (name, email, password) =>
    dispatch(addUser(name, email, password))
});

export default connect(null, mapDispatchToProps)(SignUp);

SignUp.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};
