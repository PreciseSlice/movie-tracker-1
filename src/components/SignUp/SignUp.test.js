/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { SignUp } from './SignUp';
import { shallow } from 'enzyme';

describe('SignUp', () => {
  let wrapper;
  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<SignUp handleSubmit={mockFn}
                              handleLogin={mockFn}/>)
  })

  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('has default state', () => {
    const defaultState = {
      name: '',
      password: '',
      email: '',
      error: ''
    }
    expect(wrapper.state()).toEqual(defaultState);
  })

  describe('addUserDatabase', () => {
    it('calls existingUser', () => {

    })

    it('only calls postNewUser, handleLogin, and handleSubmit if the user exists', () => {

    })

    it('alerts if email already exists', () => {

    })
  })
})
