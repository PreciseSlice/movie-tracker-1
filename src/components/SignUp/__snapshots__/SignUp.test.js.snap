// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp match snapshot 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="email"
      onChange={[Function]}
      placeholder="name"
      required={true}
      value=""
    />
    <input
      autoComplete="email"
      onChange={[Function]}
      placeholder="email"
      required={true}
      type="email"
      value=""
    />
    <input
      autoComplete="on"
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
      value=""
    />
    <button>
      Sign Up
    </button>
  </form>
</section>
`;
